test/shared.tsx

```diff
@@ -328,7 +347,23 @@ export function addInteractiveTests(render: typeof mount) {
     const allMethods = [...shouldCall, ...shouldNotCall];
 
     const wrapper = render(<Test />);
-    allMethods.forEach(method => Test.prototype[method].reset());
+    if (isNewShallowRender) {
+      allMethods.forEach(method => Test.prototype[method].resetHistory());
+    } else {
+      // Calling reset here not only resets call counts, but also any behaviors
+      // setup, such as `.returns(true)`. Calling reset here, removes our
+      // `.returns(true)` setup on shouldComponentUpdate, causing the method to
+      // return `undefined`. In shallow rendering, Enzyme spies on the return
+      // value of `sCU` and chooses to invoke lifecycle methods based on its
+      // return value. Because it returns `undefined` in this case, Enzyme skips
+      // invoking `cDU`.
+      //
+      // In the default shallow renderer, Preact invokes lifecycles. In this
+      // test, once `.reset()` is called, `.sCU` returns `undefined` and so
+      // enzyme does not invoke `cDU`. In a real test environment, cDU would be
+      // invoked twice. A simpler test that doesn't use spies can see this behavior.
+      allMethods.forEach(method => Test.prototype[method].reset());
+    }
```

Comment:
More context: The new shallow renderer does not invoke componentDidUpdate or componentDidMount. It relies on Enzyme to handle that. Without the change to `resetHistory()`, it fails this test because when `reset()` is used, `shouldComponentUpdate` returns `undefined` which Enzyme interprets to mean to not invoked `componentDidUpdate`.

If I understand this test correctly, I think the intention is to only reset the call counts and not to clear all stub returns, hence my change to call `resetHistory()`. However this breaks the test for the old shallow renderer because now it demonstrates how `componentDidUpdate` will be invoked twice (once by Preact and once by Enzyme). So I left that behavior as is for now.